### PR TITLE
Restructure NEST 3.0 guides

### PR DESCRIPTION
**Pull request by jessica-mitchell**
Due to the difficulty of navigating the very long NEST 3.0 guide, this PR re-organizes the NEST 3.0 guides by splitting it up into sections.
Basically the one file that contains the comparison between functions of NEST 2.x and NEST 3.0 remains the same
The other file was split such that
* spatially-structured related content moved to the directory guides/spatial/
* a section called NEST3.0 features includes, as separate files, new functionality of nodes, new functionality of handling connections, parametrization and recording from simulations
* The what's removed from NEST 3.0 becomes its own file as well

In addition the wording on the homepage is changed and hopefully more helpful.

There are still 2 pieces of information that have not been re-added, as I was not sure where to put them @stinebuu @hakonsbm  maybe you have some suggestions?:

1. Use allow_offgrid_times throughout: In the model ``spike_generator``, the parameter ``allow_offgrid_spikes`` is renamed                                 
  ``allow_offgrid_times`` for consistency with other models.                                                          
                                                                                                                      
2.  Use unit ms instead of number of simulation steps:The ``structural_plasticity_update_interval`` now has the unit ms instead of                                        
  number of simulation steps.                                                                                         
                              

To see how this looks, you can see it here: https://nest-test.readthedocs.io/en/separate-nest3-guides/index.html
@jougs @terhorstd Let me know if you find this easier to navigate and find information

### CODE REVIEW

**Comment by hakonsbm** [comment]
@jessica-mitchell I don't think they really fit in any of the guides. Maybe they need their own section? Or perhaps they can be below the guide links in the `All about NEST 3.0` section?

**Comment by stinebuu** [comment]
@jessica-mitchell Another option is to create a new table entry in the Reference guide: NEST 2.x vs NEST 3.0 page and place it here. I do not think it will impact that many users, so I think it would be ok to just have it in the comparison guide.

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (stinebuu) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:247]:
```suggestion
    parameter = nest.spatial_distributions.exponential(nest.spatial.distance, beta=0.15)
```

**Review by stinebuu — CHANGES_REQUESTED** [review]
Thanks @jessica-mitchell! I ended up reading through all the guides, and found that some of it is a bit outdated, so I added quite alot of comments, but they shouldn't be too hard to address I think.

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst
Comment (jessica-mitchell) [reply doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst:333]: Should this section stay on this page then? Or is there a better place for it (like maybe the parametrization.rst file?)

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/index.rst
Comment (jessica-mitchell) [reply doc/userdoc/guides/nest2_to_nest3/nest3_features/index.rst:3]: I added a little line, but I don't know what to say that hasn't been said in other pages (homepage and nest2_to_nest3/index.rst)

File: doc/userdoc/tutorials/pynest_tutorial/part_4_spatially_structured_networks.rst
Comment (jessica-mitchell) [reply doc/userdoc/tutorials/pynest_tutorial/part_4_spatially_structured_networks.rst:378]: I agree - last line was removed

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/whats_removed.rst:68]:
```suggestion
- CGConnect
- ChangeSubnet
```

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/whats_removed.rst:52]: Please add the name change of `spike_detector` to `spike_recorder`

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/whats_removed.rst:65]: I think this list would massively benefit if the items in the list were linked to related sections in `refguide_nest2_nest3` or the corresponding section of a sub-page under `nest3_features`

**Review by jougs — CHANGES_REQUESTED** [review]
Many thanks! I really think these changes improve the readability of the conversion guides a lot!

As I'm a bit short on time right now, I only browsed through the changeset and have added some comments inline whenever I noticed something. I'll try to have a second look by the end of the week. If I don't, please feel free to ping me.

File: doc/userdoc/guides/nest2_to_nest3/index.rst
Comment (sarakonradi) [inline doc/userdoc/guides/nest2_to_nest3/index.rst:13]: Should we make these more skimmable? For example:

- New features
- Spatially-structured networks (topology module)
- Old versus new function syntax
- What's removed?

Comment (sarakonradi) [inline doc/userdoc/guides/nest2_to_nest3/index.rst:13]: Should we make these more skimmable? For example:

- New features
- Spatially-structured networks (topology module)
- Old versus new function syntax
- What's removed?

**Review by sarakonradi — CHANGES_REQUESTED** [review]
@jessica-mitchell Thanks a lot for this! Just added a minor comment.

File: doc/userdoc/guides/nest2_to_nest3/index.rst
Comment (jessica-mitchell) [reply doc/userdoc/guides/nest2_to_nest3/index.rst]: Good point, done!

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jessica-mitchell) [reply doc/userdoc/guides/nest2_to_nest3/whats_removed.rst]: Links added

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jessica-mitchell) [reply doc/userdoc/guides/nest2_to_nest3/whats_removed.rst:52]: done

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst
Comment (stinebuu) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst:9]: I think many changes is better now, as people need to change to `synapse_model` and models like `spike_detector` is renamed.
```suggestion
In most use cases, you will not need to make many changes to your scripts in NEST 3.0, unless you have used **topology** or **subnets**.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst
Comment (stinebuu) [reply doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst:335]: I think this is the best page for it, parametrization.rst is slightly wrong as well. I think it is ok, I like it better now that the connect_arrays section is gone I think.

**Review by stinebuu — APPROVED** [review]
Thanks @jessica-mitchell, the guides are a lot more readable now. I just have a tiny comment, but other than that I am happy.

**Comment by jessica-mitchell** [comment]
> Thanks @jessica-mitchell, the guides are a lot more readable now. I just have a tiny comment, but other than that I am happy.

@stinebuu I also want to incorporate the random number generator docs once that PR is merged, so I'll need another review for those changes

**Comment by jessica-mitchell** [comment]
@jougs @sarakonradi @stinebuu  I added the random number generator doc into the new strucutre. Not much has changed, the section in the old overview file is now its own file in nest3_features and is included in the index.
I updated some headings to follow the style guide.

File: doc/userdoc/index.rst
Comment (stinebuu) [inline doc/userdoc/index.rst:12]: I think maybe it should be `scripts` instead of `simulations`. Maybe `simulation scripts`.

Comment (jessica-mitchell) [reply doc/userdoc/index.rst]: Done!

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst
Comment (jessica-mitchell) [reply doc/userdoc/guides/nest2_to_nest3/nest3_features/nest3_node_handles.rst]: Yes, it should have another space, I'll fix it.

**Comment by jessica-mitchell** [comment]
@jougs Could you have a final look at this PR? Thanks!

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:126]:
```suggestion
After you have created your spatially distributed nodes, you can use the `spatial` property to set
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:163]:
```suggestion
  others can be used when connecting.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:327]:
```suggestion
  |                                                    | x, the value is replaced with the value of the      |
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:334]:
```suggestion
  |     nest.math.max(x, value)                        | If a value from the parameter is below a            |
  |                                                    | threshold x, the value is replaced with the value      |
  |                                                    | of the threshold.                                   |
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:355]:
```suggestion
``max()`` functions, ``nest.math.min()`` yielding the smaller of two
values, and ``nest.math.max()`` yielding the larger of two values.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:360]:
```suggestion
    # distribution that are above 0.5 get set to 0.5.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:364]:
```suggestion
    # distribution that are below 0.5 get set to 0.5.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:368]:
```suggestion
    # distribution that are smaller than 0.2 or larger than 0.7 get
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:376]:
```suggestion
- The second argument is the resulting value or parameter evaluated if the
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:378]:
```suggestion
- The third argument is the resulting value or parameter evaluated if the
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/random_number_generators.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/random_number_generators.rst:7]:
```suggestion
threads and processes:
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/random_number_generators.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/random_number_generators.rst:55]:
```suggestion
Threefry are available out of the box.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/recording_simulations.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/recording_simulations.rst:8]:
```suggestion
also end in `_recorder`.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/recording_simulations.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/recording_simulations.rst:78]:
```suggestion
flags like `withgid`, `withtime`, `withweight`, `withport` and some
more. In many cases, this, however, lead to a confusing variety of
```

File: doc/userdoc/guides/nest2_to_nest3/refguide_nest2_nest3.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/refguide_nest2_nest3.rst:315]:
```suggestion
     The structural_plasticity_update_interval is now reported as time in ms instead of as the number of simulation steps.
```

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/whats_removed.rst:7]:
```suggestion
Subnets are gone. Instead NodeCollections should be used to group and organize neurons.
```

File: doc/userdoc/guides/nest2_to_nest3/whats_removed.rst
Comment (jougs) [inline doc/userdoc/guides/nest2_to_nest3/whats_removed.rst:83]:
```suggestion
| - DataConnect        |                              |
| - CGConnect          |                              |

```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:21]:
```suggestion
Spatially distributed nodes can now be created using the standard ``nest.Create()`` function.
```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:23]:
```suggestion
The arguments of this function have been changed to unify the
creation of populations with and without spatial information. To create
```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:86]:
```suggestion
  not supported.
```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:151]:
```suggestion
Similar to creating nodes with spatial distributions, such nodes are now connected using the
```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:173]:
```suggestion
  and ``fixed_outdegree`` implicitly only apply to either the source or target nodes.
```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:242]:
```suggestion
The former was only an alias available for backward compatibility.
```

File: doc/userdoc/guides/spatial/overview_spatial.rst
Comment (jougs) [inline doc/userdoc/guides/spatial/overview_spatial.rst:262]:
```suggestion
distributed also works, and you will receive `nan` whenever one of the nodes is non-spatial.
```

**Review by jougs — CHANGES_REQUESTED** [review]
Voilà!

**Comment by jessica-mitchell** [comment]
@jougs your suggestions have been applied! thanks

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst
Comment (jougs) [reply doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst]: Is this resolved?

**Review by jougs — APPROVED** [review]
Many thanks!

**Pull request merged**
